### DISCUSSION TITLE
Two way mirrors from one MirrorMaker2

### DISCUSSION THREAD

**Discussion by kreuzj** [strimzi/strimzi-kafka-operator#8483] [Q&A]
Hello to Strimzi,
  we are currently running bidirectional mirroring between two Kafka clusters (they are running in different AKS clusters - one AKS/Kafka for internal and second one for external purposes) using two MM2s. Each direction has own MM2 instance running in the target Kafka cluster (this arrangement should be compliant with the setup we found in your docu). Everything is working properly so far but our Customer wants us to avoid establishing the connectivity from external cluster (i.e. to avoid starting TCP session from ext cluster). The question is if it's possibile to setup two mirrors on single MM2 instance (running in the internal AKS cluster) - i.e. to have something like this setup:

apiVersion: kafka.strimzi.io/v1alpha1
kind: KafkaMirrorMaker2
spec:
  connectCluster: "int-cluster"
  clusters:
  - alias: "int-cluster"
    bootstrapServers: int-bootstrap-server
  - alias: "ext-cluster"
    bootstrapServers: ext-bootstrap-server-2

 mirrors:
  - sourceCluster: "int-cluster"
    targetCluster: "ext-cluster"
  - sourceCluster: "ext-cluster"
    targetCluster: "int-cluster"

I was already looking around and found only following discussion threads dealing with similar but not same use cases:

- https://github.com/orgs/strimzi/discussions/4979
- https://github.com/orgs/strimzi/discussions/5862
- https://github.com/orgs/strimzi/discussions/7463
- https://github.com/orgs/strimzi/discussions/4286

As some time last since those discussions I would rather ask if above describe setup would be possible. Thanks

Jan

**Comment by scholzj** [comment]
You mean to have both Mirror Maker 2 clusters running in the internal AKS cluster and connecting to the external AKS cluster from there instead of having one of them sit in the external AKS cluster and connect to the internal AKS cluster? Did I got it right?

That should be possible. The recommended way is to run the Mirror Maker 2 as close to the target cluster as possible as that helps to minimize the duplicates etc. But you can run it on the side of the source cluster as well - especially if the 2 AKS clusters are in the same region and have a good latency / networking between them.

**Reply by scholzj** [reply]
Just a note to be clear -> you would still need two separate mirror maker instances. But they can both run in the internal AKS cluster.

**Reply by kreuzj** [reply]
oki, I just though that running MM2 on the target cluster is must... Our int and ext AKS are in the same region so networking wise it should be ok. Will try to move the second instance to int. cluster ;-). Thanks for prompt reply

**Reply by scholzj** [reply]
The `connectCluster` field has to always point to the target cluster. But where it runs is more a question of how well it runs and how much it suffers from network issues. But that is more important when you mirror data across continents and not so much when its the same region.

**Reply by kreuzj** [reply]
well, it's clear now. Once again thanks for giving the right direction.
